### ISSUE TITLE
multiclass loss computation error for validation and test 

### ISSUE THREAD

**Issue by FatemehEsfahani** [NRCan/geo-deep-learning#493]
This [line](https://github.com/NRCan/geo-deep-learning/blob/develop/train_segmentation.py#L441) in the evaluation function of training pipeline gives error for multiclass loss calculation for losses such as cross entropy.
It should change to 
`loss = criterion(outputs, labels) if num_classes > 1 else criterion(outputs, labels.unsqueeze(1).float())`
The same way as it is computed in the training function.
